sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/HL7v2IO.java

```diff
@@ -179,25 +189,51 @@ public static ListHL7v2Messages readAll(List<String> hl7v2Stores) {
     return new ListHL7v2Messages(StaticValueProvider.of(hl7v2Stores), StaticValueProvider.of(null));
   }
 
+  /** Read all HL7v2 Messages from multiple stores as sendTime {@link TimestampedValue}s. */
+  public static ListTimestampedHL7v2Messages readAllWithTimestamps(List<String> hl7v2Stores) {
```

Comment:
Instead of adding static methods for each variant, any reason to not make ListHL7v2Messages be a builder so someone could invoke `withMessageTimestamps`?

This would have also make sense to make `withFilter` a builder like method on ListHL7v2Messages

Reply:
this is a good point.

Unfortunately, I think having adding replacing the existing static methods with `withFilter` builder would be interface breaking. I've added this method to `ListHL7v2Messages` as it might be more natural for some to use.

Reply:
I think you can keep the static methods that exist and transition them to use the builder underneath so you don't have to break the interface. This is done in other IOs for common scenarios.